src/Avalonia.Controls/ButtonSpinner.cs

```diff
@@ -196,13 +196,14 @@ protected override void OnKeyDown(KeyEventArgs e)
         protected override void OnPointerWheelChanged(PointerWheelEventArgs e)
         {
             base.OnPointerWheelChanged(e);
-            if (!e.Handled && AllowSpin)
+
+            if (AllowSpin && IsFocused)
```

Comment:
I think "IsKeyboardFocusWithin" is better choice here, it will return true if spinner is not focused, but any child inside of it is focused.
Also, with IsKeyboardFocusWithin you leave Focusable=true on spinner, but remove it from NumericUpDown template. So, in this way only textbox will be focusable in the NumericUpDown, but wheel scrolling will work as well only when it's focused.

Reply:
Indeed this seems to be the better fix. Also NumericUpDown seems to work better now. Please double-check if you also think it is better now,